middle_end/flambda2/types/provers.ml

```diff
@@ -376,63 +429,51 @@ let prove_variant_like_generic env t : variant_like_proof generic_proof =
               | Proved const_ctors -> Known const_ctors)
           in
           Proved { const_ctors; non_const_ctors_with_sizes })))
-  | Value (Ok (Mutable_block _)) -> Unknown
-  | Value (Ok (Array _)) ->
-    Unknown (* We could return Invalid in a strict mode *)
-  | Value
-      (Ok
-        ( Closures _ | Boxed_float _ | Boxed_float32 _ | Boxed_int32 _
-        | Boxed_int64 _ | Boxed_vec128 _ | Boxed_nativeint _ | String _ )) ->
+  | Mutable_block _ -> Unknown
+  | Array _ -> Unknown (* We could return Invalid in a strict mode *)
+  | Closures _ | Boxed_float _ | Boxed_float32 _ | Boxed_int32 _ | Boxed_int64 _
+  | Boxed_vec128 _ | Boxed_nativeint _ | String _ ->
     Invalid
-  | Value Unknown -> Unknown
-  | Value Bottom -> Invalid
-  | Naked_immediate _ | Naked_float _ | Naked_float32 _ | Naked_int32 _
-  | Naked_int64 _ | Naked_vec128 _ | Naked_nativeint _ | Rec_info _ | Region _
-    ->
-    wrong_kind "Value" t
 
 let meet_variant_like env t =
-  as_meet_shortcut (prove_variant_like_generic env t)
+  gen_value_to_meet prove_variant_like_generic_value env t
 
-let prove_variant_like env t = as_property (prove_variant_like_generic env t)
+let prove_variant_like env t =
+  gen_value_to_proof prove_variant_like_generic_value env t
 
 type boxed_or_tagged_number =
   | Boxed of
       Alloc_mode.For_types.t * Flambda_kind.Boxable_number.t * Type_grammar.t
   | Tagged_immediate
 
 (* CR pchambart: Remove fragile matchs and reuse this function *)
```

Comment:
This comment could now be removed I think.

Reply:
Done